### PR TITLE
Integration Test Setup

### PR DESCRIPTION
**Pull request by aksh0001**
# Description 

Setup integration testing functionality for Researchify's Express API server. The tests are intended to validate/verify the API's endpoints.

# Type of change 
- [ ] Bug fix
- [x] New feature
- [ ] Refactoring
- [x] Documentation 

# Problem

Our previous tests were empty/non-functional. We didn't test any real functionality. They were serving as a placeholder for real tests.

# Solution description

Implement API endpoint testing using `jest` and `supertest`. The details are explained in the `README` document that I created along with the code changes.

Note: to make the tests run on the CI, I've modified our workflow script to start up MongoDB as part of the testing stage.

# Tests 
All unit tests pass? Y

Builds successfully? Y

Provide steps in detail for reviewers to reproduce the expected result of the code changes.

To verify, please install MongoDB locally and run the tests yourselves. Also please check for any regressions as I've modified the API source code a little  (specifically server setup and startup) to make the tests work.

# Relevant document/ link (if any) 
See the provided `README`.

# Risk
Low, additive.

### CODE REVIEW

File: api/.eslintrc.json [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
-    }
+  "env": {
+    "commonjs": true,
+    "es2021": true,
+    "node": true
+  },
+  "extends": [
+    "eslint:recommended",
+    "airbnb-base",
+    "plugin:jest/recommended"
```
Comment (aksh0001) [inline api/.eslintrc.json:10]: This new extension will allow eslint to run on our tests.

**Review by ssha0054 — APPROVED** [review]
LGTM! Thanks for providing a detailed `readme` with useful links which make it easier to understand and run the tests. I was initially getting an error when running the tests and then realised I wasn’t running mongo locally before starting the tests. Had a few issues trying to get mongo to run locally but once that was resolved the tests run fine. Once this PR gets merged, it should be a lot easier to create tests on the other routes now.

**Comment by aksh0001** [comment]
Note: all status checks have passed. The yellow indicator is saying that `Unit Tests` need to pass, but I've renamed the stage to `Integration Tests`. This is a GitHub bug :(

**Review by jfan0010 — APPROVED** [review]
Its working on mine.
Friendly reminder for others: rmb to `npm install` in `./api`

File: api/test/routes/homepage.test.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  await seedDb();
+});
+
+afterEach(async () => {
+  await clearAllCollections();
+});
+
+describe('POST /homepage/:team_id', () => {
+  it('should return 404 for a non-existent team', async () => {
+    const res = await request.post(`${ROUTE_PREFIX}/613b888ffca059539f01fc64`);
```
Comment (jfan0010) [inline api/test/routes/homepage.test.js:42]: Dumb question: Why are we specifying this particular object id? Will MongoDB never generate this id when we're creating the database for this endpoint?

Comment (aksh0001) [reply api/test/routes/homepage.test.js:42]: Good Q.  I'm counting on this to be unique enough that we never generate this ID again. We need to meet the hexadecimal and length requirements so just chose an arbitrary id that's long enough.

**Pull request merged**